sklearn/datasets/_openml.py

```diff
@@ -134,21 +150,14 @@ def _get_json_content_from_openml_api(url, error_message, raise_if_error,
         (acceptable error is, e.g., data id not found. Other errors, like 404's
         will throw the native error message)
 
-    raise_if_error : bool
-        Whether to raise an error if OpenML returns an acceptable error (e.g.,
-        date not found). If this argument is set to False, a None is returned
-        in case of acceptable errors. Note that all other errors (e.g., 404)
-        will still be raised as normal.
```

Comment:
Remove it and catch the exception directly in one case when it's necessary. This makes this function return a single type, instead of Dict or None.